### PR TITLE
Stubs for classic mgmt ioctls

### PR DESCRIPTION
**Pull request by sonals**

### CODE REVIEW

File: src/include/xocl-subdev.h [hunk 1]
```diff
@@ -155,7 +156,8 @@ enum xocl_events {
 	XOCL_EVENT_PRE_REMOVAL,
 
 	/* Broadcast'able events from leaf. */
-	XOCL_BROADCAST_EVENT_TEST
+	XOCL_BROADCAST_EVENT_TEST,
+	XOCL_BROADCAST_EVENT_MGMT
```
Comment (maxzhen) [inline src/include/xocl-subdev.h:160]: Hmm...what is this for?

Comment (sonals) [reply src/include/xocl-subdev.h]: No current use, removed.

File: src/lib/xocl-main.c [hunk 2]
```diff
@@ -125,11 +133,52 @@ static void xocl_drv_unregister_driver(enum xocl_subdev_id id)
...
+	int result = 0;
+
+	mutex_lock(&xocl_class_lock);
+	for (i = 0; i < ARRAY_SIZE(xocl_drv_maps); i++) {
+		struct xocl_drv_map *map = &xocl_drv_maps[i];
+
+		if (map->id != id)
+			continue;
+		if (map->drv) {
+			result = -EEXIST;
```
Comment (maxzhen) [inline src/lib/xocl-main.c:148]: print out a warning or just bug_on?

File: src/lib/xocl-main.c [hunk 3]
```diff
@@ -125,11 +133,52 @@ static void xocl_drv_unregister_driver(enum xocl_subdev_id id)
 	pr_info("unregistered %s subdev driver\n", drvname);
 }
 
+int xocl_subdev_register_external_driver(enum xocl_subdev_id id, struct platform_driver *drv)
```
Comment (maxzhen) [inline src/lib/xocl-main.c:136]: after you fill in the platform_driver pointer, i think you also need to call xocl_drv_register_driver for this one as well?

File: src/lib/xocl-main.c [hunk 4]
```diff
@@ -125,11 +133,52 @@ static void xocl_drv_unregister_driver(enum xocl_subdev_id id)
...
+		}
+		map->drv = drv;
+	}
+out:
+	mutex_unlock(&xocl_class_lock);
+	return 0;
+}
+EXPORT_SYMBOL_GPL(xocl_subdev_register_external_driver);
+
+void xocl_subdev_unregister_external_driver(enum xocl_subdev_id id)
```
Comment (maxzhen) [inline src/lib/xocl-main.c:159]: also needs to call xocl_drv_unregister_driver() inside?

File: src/mgmt/xmgmt-root.c [hunk 5]
```diff
@@ -11,6 +11,7 @@
 #include <linux/module.h>
 #include <linux/pci.h>
 #include <linux/aer.h>
+#include <linux/vmalloc.h>
```
Comment (maxzhen) [inline src/mgmt/xmgmt-root.c:14]: what is this header for? i don't see any code change...

File: src/lib/xocl-main.h [hunk 6]
```diff
@@ -11,6 +11,8 @@
 
 extern struct platform_driver xocl_partition_driver;
 extern struct platform_driver xocl_test_driver;
+extern struct platform_driver xocl_mgmt_driver;
+extern struct platform_driver xmgmt_main_driver;
```
Comment (maxzhen) [inline src/lib/xocl-main.h:15]: should remove them?

File: src/lib/xocl-main.c (hunk 2 above)
Comment (sonals) [reply src/lib/xocl-main.c:153]: Updated to print error message.

File: src/lib/xocl-main.c (hunk 3 above)
Comment (sonals) [reply src/lib/xocl-main.c:141]: Done

File: src/lib/xocl-main.c (hunk 4 above)
Comment (sonals) [reply src/lib/xocl-main.c:165]: Done

File: src/mgmt/xmgmt-root.c (hunk 5 above)
Comment (sonals) [reply src/mgmt/xmgmt-root.c:14]: Needed to compile on POWER9 :-)

File: src/lib/xocl-main.h (hunk 6 above)
Comment (sonals) [reply src/lib/xocl-main.h]: Done

**Pull request merged**